components/Heading/Heading.tsx

```diff
@@ -0,0 +1,49 @@
+// Copyright (c) Gridiron Survivor.
+// Licensed under the MIT License.
+
+import React, { forwardRef, HTMLAttributes } from 'react';
+
+interface HeadingProps extends HTMLAttributes<HTMLElement> {
+  heading?: keyof typeof headingVariants;
+}
+
+const headingVariants = {
+  h1: {
+    className: 'font-extrabold text-5xl leading-none tracking-tight',
+  },
+  h2: {
+    className: 'font-bold text-[2rem] leading-[1.125rem] tracking-tighter',
+  },
+  h3: {
+    className: 'font-semibold text-base leading-8 tracking-tighter"',
+  },
+  h4: {
+    className: 'font-semibold text-sm leading-7 tracking-tighter',
+  },
+};
+
+/**
+ * Heading component with dynamic heading level and styling.
+ *
+ * @param {HeadingProps} props - The props for the Heading component.
+ * @param {keyof typeof headingVariants} props.heading - The heading level. Defaults to 'h1'.
+ * @param {string} props.className - Additional CSS class name for the heading.
+ * @param {HTMLAttributes<HTMLElement>} props.rest - Additional HTML attributes for the heading.
+ * @returns {React.ReactElement} The rendered Heading component.
+ */
+const Heading = forwardRef<HTMLElement, HeadingProps>(
```

Comment:
Unless I'm not understanding it properly, how is this component putting out the proper heading `h1`, `h2`, `h3`, etc. based on the variant specified? 

To my understanding, we want the component to output the proper heading element _alongside_ the proper styles.

Reply:
This is correct. It's not rendering the HTML web standards elements. H1 should use `<h1>` with the h1 class names you have created. But, the common classes should be in the default classname.

Reply:
This issue still exists. An H1 should render`<h1>HELLO</h1>`